src/components/styles/Spacing.ts

```diff
@@ -0,0 +1,82 @@
+import { css } from 'styled-components';
+import { spacing as sizes } from '../../constants/spacing';
+import grid, { TGridBreakpoints } from '../../constants/grid';
+
+type TSpacingTypes = 'margin' | 'padding';
+
+const createTopLevelSizes = (type: TSpacingTypes) =>
+  Object.keys(sizes).reduce(
```

Comment:
there is `0` breakpoint so you shouldn't need this part actually

Reply:
My bad I didn't realise there was a zero breakpoint thanks

Reply:
Looking at it closer I prefer p-2 over `xs:p-2` I am going to filter zero out of the breakpoints